src/DSharp.Compiler/ScriptModel/Symbols/SymbolSet.cs

```diff
@@ -952,48 +952,92 @@ private TypeSymbol ResolveAtomicNameNodeType(AtomicNameNode atomicNameNode, ISym
             return null;
         }
 
-        public void AddExtensionType(string typeToExtend, string extensionMethodName, MethodSymbol methodSymbol)
+        public void AddExtensionMethod(MethodSymbol extensionMethod)
         {
-            if (string.IsNullOrWhiteSpace(typeToExtend) || string.IsNullOrWhiteSpace(extensionMethodName))
+            if (!extensionMethods.ContainsKey(extensionMethod.Name))
             {
-                return;
+                extensionMethods[extensionMethod.Name] = new List<MethodSymbol>();
             }
 
-            if (!extensionMethods.TryGetValue(typeToExtend, out HashSet<(string method, MethodSymbol relatedType)> registrations))
-            {
-                registrations = new HashSet<(string method, MethodSymbol relatedType)>();
-                extensionMethods.Add(typeToExtend, registrations);
-            }
-
-            registrations.Add((extensionMethodName, methodSymbol));
+            extensionMethods[extensionMethod.Name].Add(extensionMethod);
         }
 
         //TODO: Migrate this to be on the symbol directly
         public MethodSymbol ResolveExtensionMethodSymbol(TypeSymbol type, string memberName)
         {
-            var extensionMethods = TypeSymbolVisitor.Visit(type, t => GetTypeExtensionMethod(t, memberName))
+            //Check none generic version
+            var extensionMethods = TypeSymbolVisitor.Visit(type, t => FindExtensionMethodWithNoneGenericTarget(t, memberName))
                 .Where(i => i != null)
                 .Distinct()
                 .ToList();
 
-            return extensionMethods.Count == 1
-                ? extensionMethods.First()
-                : null;
+            if (extensionMethods.Any())
+            {
+                return extensionMethods.First();
+            }
+
+            //Check none constraint generic version
+            extensionMethods = TypeSymbolVisitor.Visit(type, t => FindExtensionMethodWithGenericTarget(t, memberName))
+                .Where(i => i != null)
+                .Distinct()
+                .ToList();
+
+            if (extensionMethods.Any())
+            {
+                return extensionMethods.First();
+            }
+
+            var objectType = new ClassSymbol("System.Object", SystemNamespace);
+            return FindExtensionMethodWithNoneGenericTarget(objectType, memberName) ?? FindExtensionMethodWithGenericTarget(objectType, memberName);
         }
 
-        private MethodSymbol GetTypeExtensionMethod(TypeSymbol type, string memberName)
+        private MethodSymbol FindExtensionMethodWithNoneGenericTarget(TypeSymbol type, string name)
```

Comment:
`FindExtensionMethodWithNoneGenericTarget` -> `FindExtensionMethodWithNonGenericTarget`

Reply:
Just need to remove the e :D